### PR TITLE
Update optee_utils.c to fix ARM-software/tf-issues#603

### PR DESCRIPTION
**Pull request by nathan-menhorn**
Updated the code to fix issue https://github.com/ARM-software/tf-issues/issues/603 related to the tee-validate-header bug.

### CODE REVIEW

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

**Comment by danh-arm** [comment]
minor: The code looks good but the commit message and PR description incorrectly references https://github.com/ARM-software/arm-trusted-firmware/pull/603 in the main repo instead of https://github.com/ARM-software/tf-issues/issues/603 in the tf-issues repo.

**Comment by nathan-menhorn** [comment]
Commit message updated.

**Comment by danh-arm** [comment]
You've updated the PR description but not the commit message.

**Comment by soby-mathew** [comment]
@nathan-menhorn , the commit message of the patch needs to be updated from
 "Updated the code to fix issue #603 related to the tee-validate-header bug."
to
"Updated the code to fix ARM-software/tf-issues#603 related to the tee-validate-header bug.

Fixes  ARM-software/tf-issues#603
"

File: lib/optee/optee_utils.c [hunk 1]
```diff
@@ -26,11 +26,7 @@ typedef struct optee_image {
 #define OPTEE_PAGER_IMAGE_ID		0
 #define OPTEE_PAGED_IMAGE_ID		1
 
-/** Minimum number of binary images in order to validate the header. */
-#define OPTEE_MIN_IMAGE_NUM		1u
-
-/** Maximum number of binary images in order to validate the header. */
-#define OPTEE_MAX_IMAGE_NUM		2u
+#define OPTEE_MAX_NUM_IMAGES		2u
```
Comment (soby-mathew) [inline lib/optee/optee_utils.c]: nit:  Could you squash this commit with the first patch?

**Comment by nathan-menhorn** [comment]
Updated. I missed the --force while pushing when rebasing the first time. This is now fixed.

**Comment by antonio-nino-diaz-arm** [comment]
Is this PR ready?

The only thing I can think of is that the commit title is too long, it should be ~50 characters long max.

**Comment by nathan-menhorn** [comment]
@antonio-nino-diaz-arm The PR is ready and I just condensed the commit message title and cleaned up the commit message itself.

**Comment by soby-mathew** [comment]
@nathan-menhorn, Could you fix the commit message to be within 80 column width and also address the comment by @etienne-lms ?

**Comment by antonio-nino-diaz-arm** [comment]
And now that @soby-mathew mentions it, you are missing the Signed-off-by line,

**Comment by nathan-menhorn** [comment]
Will add the signed-off by line and fix the columns. I will leave the valid comment since there are several checks in this function and makes clear that valid applies to the entire header and not just a particular section.

**Comment by nathan-menhorn** [comment]
Updated.

**Comment by soby-mathew** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
Could you rebase this patch on origin/master and run checkpatch.pl to fix the ERRORs?

**Comment by soby-mathew** [comment]
@nathan-menhorn, the patch will be good to merge once the white-space and coding style errors flagged by checkpatch.pl  are corrected.

**Comment by nathan-menhorn** [comment]
I’ll work on all the feedback.

On Aug 18, 2018, at 4:05 AM, Etienne Carriere <notifications@github.com<mailto:notifications@github.com>> wrote:


EXTERNAL EMAIL

@etienne-lms commented on this pull request.

@nathan-menhorn<https://github.com/nathan-menhorn>, as you need to update style for this change, please consider my comment about in-line comments.

________________________________

In lib/optee/optee_utils.c<https://github.com/ARM-software/arm-trusted-firmware/pull/1484#discussion_r211071365>:

> @@ -53,14 +64,29 @@ typedef struct optee_header {
  ******************************************************************************/
 static inline int tee_validate_header(optee_header_t *header)
 {
+       /* Indicates whether or not the image header is valid. */


quite minor: this comment is useless. should remove it and the others also. If each sequence is explained by a inserted comment, code becomes ugly.
This function API is already described at line 51<https://github.com/ARM-software/arm-trusted-firmware/pull/1484/files#diff-5983196625ba2a89d7c638bff6295463R51>. Maybe simply:

        if ((header->magic == TEE_MAGIC_NUM_OPTEE) &&
            (header->version == 2) &&
            (header->nb_images > 0u) &&
            (header->nb_images <= OPTEE_MAX_IMAGE_NUM)) {
                return 1;
        }

        WARN("Not a known TEE, use default loading options.\n");
        return 0;

**Comment by nathan-menhorn** [comment]
checkpatch.pl issues fixed. Code is checked in and rebased.

**Comment by soby-mathew** [comment]
jenkins: test this please

**Comment by soby-mathew** [comment]
jenkins: test this please

**Pull request merged**